Versión inicial para iniciar el desarrollo del proyecto SchoolBook

jorgerdc opened the pull request:
Se define el caso de estudio. Esta es una. primera versión la cual se irá refinando conforme se avance en el diseño.
La idea es generar un caso de estudio sencillo de entender para poder ser implementado empleando principios básicos de la arquitectura de microservicios.

jorgerdc committed:
Versión inicial para iniciar el desarrollo del proyecto SchoolBook

alexJava2015 commented on README.md:
```diff
@@ -1,2 +1,27 @@
+* Instalación de la IDE de desarrollo [Spring Tool Suite](https://spring.io/tools)
+* Uso del algún editor de texto, de preferencia [Visual Studio Code](https://code.visualstudio.com/).
+## Documentación y diseño
+* Revisar la carpeta docs en la que se encuentran los siguientes documentos:
```
Corregir la descripción de este punto, ya que se mencionan archivos que aun no existen.

jorgerdc replied:
Corregido

alexJava2015 replied:
Perfecto se cierra comentario.

alexJava2015 commented on README.md:
```diff
@@ -1,2 +1,27 @@
+	* Diseño de bases de datos y esquemas (modelos relacionales).
+## Código fuente
+* El código fuente del proyecto se distribuye en diversos repositorios.
+	* [schoolbook](https://github.com/jorgerdc/schoolbook) (este repositorio)  Contiene la definición del proyecto, documentación, así como el código de la aplicación web.
```
Cambiar la ruta por una valida, ya que esta no existe.

jorgerdc replied:
Resuelto.

alexJava2015 commented on docs/caso-estudio.md:
```diff
@@ -0,0 +1,108 @@
+#### Control  de libros en bibliotecas
+* Cuando un libro es solicitado o asignado a las bibliotecas se debe registrar en base de datos esta asociación,  su número de clasificación, así como la fecha de asignación. Estos datos son independientes al tipo de libro.
+* Para el caso de los libros físicos se almacena el número total de ejemplares con los que cuenta la biblioteca. Este número de ejemplar se incluye en una etiqueta junto al número de clasificación que se pega en el lomo del ejemplar.
+* Para cada ejemplar se registra el número de ejemplar iniciando en 0001 y la ubicación dentro de la biblioteca. 
```
Mejorar la redacción para no repetir la palabra ejemplar.

jorgerdc replied:
Se modifica la redacción.

alexJava2015 replied:
Perfecto se cierra comentario.

alexJava2015 commented on docs/caso-estudio.md:
```diff
@@ -0,0 +1,108 @@
+* Para determinar la ubicación del ejemplar dentro de la biblioteca se emplea una cadena con el siguiente formato.  Suponer que un libro se encuentra en el piso 3, sala norte, sección M,  anaquel 5,  renglón 3.  Su cadena será:  `PS3-SAN-SEM-AN5-R3` . Cada biblioteca puede definir el formato de esta cadena.
+* Para llevar el control de los libros se ha diseñado una serie de status.
+
+|Status            |Descripción
```
Pasar a español la palabra "status".

jorgerdc replied:
Considero que debería quedarse así. La traducción en español sería "Estado" la cual es un tanto ambigua, por ejemplo, se puede confundir con "Estados de la república", etc.
En documentos formales ya se vale que existan palabras en inglés cuando su traducción provoque confusión

alexJava2015 replied:
De acuerdo, con esta aclaración, queda en inglés la palabra. se cierra comentario.

alexJava2015 commented on docs/caso-estudio.md:
```diff
@@ -0,0 +1,108 @@
+| ---              | ---------- 
+|EN ALMACEN        | El ejemplar se encuentra en almacén
+|EN BIBLIOTECA     | El ejemplar se ha distribuido a una biblioteca
+|EN LINEA          | EL libro digital está disponible para ser accedido vía internet
```
Corregir la letra "L" pasarla a minúscula dentro del texto "EL libro ...."

jorgerdc replied:
Corregido

alexJava2015 replied:
Se cierra comentario.

alexJava2015 commented on docs/caso-estudio.md:
```diff
@@ -0,0 +1,108 @@
+CON MULTA   | El Usuario aun no ha realizado la entrega del préstamo, tiene al menos un ejemplar con fecha de entrega expirada y han pasado los 2 días que el sistema otorga para regresar los ejemplares. El importe de la multa a pagar se establecerá una vez que el usuario regrese los ejemplares a la biblioteca.
+ENTREGADO  | El usuario ha devuelto todos los ejemplares a la biblioteca en buen estado y sin contar con fechas de entrega expiradas.
+ENTREGADO CON MULTA | El usuario ha devuelto los ejemplares, pero se detecta que el usuario debe pagar multa por alguno de los siguientes escenarios: 1. El usuario no realizó la entrega de sus ejemplares en las fechas establecidas.  2. El usuario regresó los ejemplares con daños o no los regresa por pérdida.  En el momento de la entrega, se asigna la multa al usuario. Dependiendo el daño, el oper ...
+* Se requiere llevar el control de un histórico de estos status.
```
Pasar a español la palabra "status" y que todo el texto  del documento esta así.

jorgerdc replied:
Mismo comentario respecto al caso anterior.

alexJava2015 replied:
De acuerdo, con la aclaración, queda en inglés la palabra. se cierra comentario.

alexJava2015 commented on docs/caso-estudio.md:
```diff
@@ -0,0 +1,108 @@
+* Para el caso de los libros digitales, se habilita una clave de acceso de 5 caracteres que será empleada para acceder al contenido del libro desde internet.  Una vez que la fecha de entrega del libro ha expirado, el sistema ya no permitirá acceder a su contenido.
+* De forma similar a los libros, se lleva un control de status de préstamos
+
+Status  |Descripción  
```
Pasar a español la palabra "status" y que todo el texto del documento esta así.

jorgerdc replied:
Mismo comentario respecto al caso anterior.

alexJava2015 replied:
De acuerdo, con la aclaración, queda en inglés la palabra. se cierra comentario.

alexJava2015 commented on docs/caso-estudio.md:
```diff
@@ -0,0 +1,108 @@
+* Se requiere llevar el control de un histórico de estos status.
+
+#### Surtido de libros para bibliotecas.
+* Las bibliotecas pueden solicitar a SchooBook el surtido ya sea de libros existentes o de nuevos libros.  Para ello se requiere registrar una solicitud de surtido.   Se almacena un folio de solicitud, fecha de solicitud,  usuario que realiza el pedido (generalmente el administrador de la biblioteca).
```
Corregir la palabra "SchooBook", falta letra "l" en la palabra "School"

jorgerdc replied:
corregido

alexJava2015 replied:
Perfecto se cierra comentario.

alexJava2015 commented on docs/caso-estudio.md:
```diff
@@ -0,0 +1,108 @@
+* El sistema web ofrece la funcionalidad clásica del *carrito de compras*.
+* Nuevamente, el control de una orden de compra se realiza a través de la aplicación de una serie de estados.
+
+Status | Descripción
```
Pasar a español la palabra "status" y que todo el texto del documento esta así.

jorgerdc replied:
Mismo comentario que el caso anterior.

alexJava2015 replied:
De acuerdo, con la aclaración, queda en inglés la palabra. se cierra comentario.

rodrigofrancisco commented on docs/caso-estudio.md:
```diff
@@ -0,0 +1,108 @@
+* Escuelas en general contratan los servicios de *SchoolBook* para llevar el control de  sus bibliotecas. 
+* La escuela debe solicitar sus servicios a través de un contrato que contiene los siguientes datos: número de contrato, fecha inicio y fin del contrato, nombre de la escuela o institución, clave escolar,  responsable del contrato (generalmente el director de la escuela).
+* En el detalle del contrato también se especifica la lista de bibliotecas que serán administradas por  SchoolBook.  Para cada biblioteca se registra la siguiente información:  Nombre, dirección,  ubicación geográfica (latitud, longitud), dirección de su página web, número aproximado de ejemplares que maneja,  número aproximado de usuarios que maneja.
+* Con base a estos datos, en el contrato se especifica la cuota mensual que la escuela deberá pagar.   Una  vez que el contrato es firmado, se obtiene una copia en formato PDF el cual se almacena en base de datos.
```
Debería decir: "Con base en estos datos" ...

jorgerdc replied:
corregido

rodrigofrancisco commented on docs/caso-estudio.md:
```diff
@@ -0,0 +1,108 @@
+| EN PRESTAMO CON RETARDO  | El ejemplar ha sido prestado a un usuario, pero el periodo para ser regresado a la biblioteca ha expirado
+| EN REPARACIÓN | El ejemplar presenta daños que pueden ser reparados. Mientras el ejemplar no sea reparado permanecerá en este estado.
+* Se requiere contar con histórico de status.
+#### Control de prestamos
+* Los prestamos aplican a ambos tipos de libros (ejemplares y digitales).
```
Acento en "prestamos" tanto en el título como al inicio del primer párrafo

jorgerdc replied:
corregido

rodrigofrancisco commented on docs/caso-estudio.md:
```diff
@@ -0,0 +1,108 @@
+
+Status  |Descripción  
+ ---        | --- 
+VIGENTE     | El préstamo se ha otorgado y las fechas de entrega de cada uno de sus ejemplares aun no expiran.
```
"aun" debería llevar tilde no?

jorgerdc replied:
corregido

rodrigofrancisco commented on docs/caso-estudio.md:
```diff
@@ -0,0 +1,108 @@
+ ---        | --- 
+VIGENTE     | El préstamo se ha otorgado y las fechas de entrega de cada uno de sus ejemplares aun no expiran.
+EXPIRADO    | Por lo menos una de las fechas de entrega de los ejemplares del préstamo ha expirado sin haber sido entregado a la biblioteca.  Cuando esto ocurre, el usuario se le otorgarán 2 días adicionales antes de establecer una multa.
+CON MULTA   | El Usuario aun no ha realizado la entrega del préstamo, tiene al menos un ejemplar con fecha de entrega expirada y han pasado los 2 días que el sistema otorga para regresar los ejemplares. El importe de la multa a pagar se establecerá una vez que el usuario regrese los ejemplares a la biblioteca.
```
La "U" mayúscula de Usuario es intencional la identificar al actor "usuario"? Si no se sugiere cambiarla a minúscula

rodrigofrancisco replied:
Colocar la tilde de "aun"

jorgerdc replied:
corregido.

rodrigofrancisco commented on docs/caso-estudio.md:
```diff
@@ -0,0 +1,108 @@
+* Adicional a los datos que se guardan de cada libro, se guarda su precio el cual es aplicable en cierto periodo de tiempo. 
+* Los usuarios que desean comprar libros deben registrar los datos de al menos una tarjeta de crédito (número, tipo, año y mes de expiración, número de seguridad).   Se requiere registrar de forma obligatoria al menos una dirección de entrega (calle, número, código postal , municipio/delegación/alcaldía/ciudad,  estado y país).
+* Cabe mencionar que los usuarios registrados que pertenecen a las escuelas mencionadas anteriormente, pueden hacer uso de sus datos para realizar compras en el sitio web.  Solo deberán registrar los datos de su tarjeta y sus direcciones de envío.
+* Para usuarios que no pertenecen a las escuelas, estos no podrán acceder a los prestamos  que son ofrecidos por las escuelas.
```
acento en "prestamos"

jorgerdc replied:
corregido

jorgerdc committed:
Corrección de comentarios de la primera revisión.
Adicionalmente se agrega detalle y se refinan algunas reglas de negocio
que permitan definir adecuadamente los modelos de bases de datos.

jorgerdc committed:
se corrige sintaxis de formato. La última parte del archivo no se estaba mostrando correctamente.

jorgerdc committed:
se corrige sintaxis de formato. La última parte del archivo no se estaba mostrando correctamente.

jorgerdc commented:
@rhofp please, podrías echarle un ojo rápido al segundo update?  Este merge es importante para el curso.

rodrigofrancisco commented on docs/caso-estudio.md:
```diff
@@ -1,89 +1,95 @@
+* En el detalle del contrato también se especifica la lista de bibliotecas que serán administradas por  SchoolBook.  Para cada biblioteca se registra la siguiente información:  Nombre, dirección,  ubicación geográfica (latitud, longitud), dirección de su página web,  teléfonos de contacto, número aproximado de ejemplares que maneja,  número aproximado de usuarios que maneja.
+* Con base en estos datos, en el contrato se especifica la cuota mensual que la escuela deberá pagar para cada una de las bibliotecas.   Una  vez que el contrato es firmado, se obtiene una copia en formato PDF el cual se almacena en base de datos.
 * La escuela puede renovar su contrato las veces que así lo desee. En realidad renovar un contrato significa generar uno nuevo con las mismas condiciones y asociarlo con el contrato anterior. La cuota mensual puede cambiar al renovar.
+* Para realizar los pagos mensuales, la escuela puede proporcionar varios números de tarjeta de crédito, puede ser inclusive a nivel de biblioteca. Se registra el número de tarjeta, su tipo, mes y año de expiración y número de seguridad.
```
Cuando dices que "puede ser a nivel de biblioteca" se refiere a que es posible registrar una tarjeta por biblioteca, cierto? Por que me parece que se puede malinterpretar como que la entidad biblioteca puede registrar "n" tarjetas de crédito, lo cual creo que es falso.

jorgerdc replied:
Es correcto.  Una escuela puede registrar varias tarjetas, pero  para cada biblioteca solo se asigna una. Modificaré redacción para aclarar este punto  👍

rodrigofrancisco commented on docs/caso-estudio.md:
```diff
@@ -1,89 +1,95 @@
-* Cada libro digital se almacena en varios archivos. Se almacena el archivo, el rango de páginas que contiene cada archivo, nombre del archivo, formato y tamaño en bytes.
-* Algunos libros digitales incluyen material multimedia, archivos de audio y/o archivos de video.  Para cada archivo se registra un número de archivo, su formato y una descripción.
+* Cada libro digital se almacena en varios archivos. Se almacena el archivo,  su número iniciando en 1 por cada libro, el rango de páginas que contiene cada archivo, nombre del archivo, formato y tamaño en bytes.
+* Algunos libros digitales incluyen material multimedia, archivos de audio y/o archivos de vídeo.  Para cada archivo se registra un número de archivo,  su tipo (vídeo, audio, etc.), su formato y una descripción.
```
Ok, entonces tipo describe si es vídeo, audio, etc. y el formato describe si es jpg, mp3, etc. cierto? 
Creo que sería buena idea específicarlo por si algún despistado como yo se confunde en eso. Porque en el commit anterior había entendido que el tipo era jpg, png,...

jorgerdc replied:
ok de acuerdo.

rodrigofrancisco reviewed (changes requested):
Listo @jorgerdc

jorgerdc committed:
Se actualiza documento con base a comentarios de revisión

alexJava2015 commented on README.md:
```diff
@@ -1,2 +1,30 @@
+	* Servicio encargado de la administración de préstamos y ejemplares.
+	* Servicio encargado de administrar las  ordenes de compra de libros.
+## Código fuente
+* El código fuente del proyecto se distribuye en diversos repositorios los cuales se pueden consultar [en este enlace.]([https://github.com/school-book/](https://github.com/school-book/)) 
```
No se muestra ningún enlace, favor de corregir.

jorgerdc committed:
Corrección de un link mal formado
